pkg/controllers/workgenerator/controller.go

```diff
@@ -197,7 +204,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 	// update the resource binding status
 	if updateErr := r.Client.Status().Update(ctx, &resourceBinding); updateErr != nil {
 		klog.ErrorS(updateErr, "Failed to update the resourceBinding status", "resourceBinding", bindingRef)
-		return controllerruntime.Result{}, controller.NewUpdateIgnoreConflictError(updateErr)
+		return controllerruntime.Result{Requeue: true}, controller.NewUpdateIgnoreConflictError(updateErr)
```

Comment:
why requeue when there is an error?

Reply:
A requeue is need when there is an error. ASAIK, the error is expected and we continue on. Without this requeue, I do not see the available condition updated.

Reply:
Tested it locally. Works without it.